include/nuttx/list.h

```diff
@@ -275,28 +234,26 @@
  * node and temp_node should be struct list_node*
  */
 
-#define list_for_every_safe(list, node, temp_node) \
-  for(node = (list)->next, temp_node = (node)->next; \
-      node != (list); \
-      node = temp_node, temp_node = (node)->next)
+#define list_for_every_safe(list, node, temp) \
+  for(node = (list)->next, temp = node->next; \
+      node != (list); node = temp, temp = node->next)
 
 /* iterates over the list, entry should be the container structure type */
 
 #define list_for_every_entry(list, entry, type, member) \
-  for((entry) = container_of((list)->next, type, member); \
-      &(entry)->member != (list); \
-      (entry) = container_of((entry)->member.next, type, member))
+  for(entry = container_of((list)->next, type, member); \
```

Comment:
Why this is changed?

Reply:
entry is different from other macro argument since it must be a variable(lvalue), so () is  redundant.